### DISCUSSION TITLE
IPSEC site-to-site with Cisco ASA phase 2 ECP_521 strange issue

### DISCUSSION THREAD

**Discussion by cpir0** [strongswan/strongswan#627] [Q&A] [answered]
Hi all,

We have a site-to-site established between a cisco ASA device, see config below,

conn xxx-xxx-xxx
      auto=start
      type=tunnel
      left=xx.xx.xx.xx
      leftsubnet=xx.xx.xx.xx/32
      leftid=xx.xx.xx.xx
      right=xx.xx.xx.xx
      rightsubnet=xx.xx.xx.xx/30
      authby=psk
      keyexchange=ikev2
      ike=aes256-sha256-modp2048
      esp=aes256-sha256-ecp521
      mobike=no
      keyingtries=3     
      ikelifetime=24h
      lifetime=1h
      dpddelay=100s
      dpdtimeout=300s
      dpdaction=restart

While rekeying phase2 we get the following weird log message and rekeying does not go through,

08[ENC] parsed CREATE_CHILD_SA response 78 [ N(INVAL_KE) ]
08[IKE] peer didn't accept DH group ECP_521, it requested ECP_521
08[IKE] establishing CHILD_SA xx-xxx-xxx{44459} reqid 709
08[ENC] generating CREATE_CHILD_SA request 79 [ N(REKEY_SA) SA No TSi TSr ]
08[NET] sending packet: from xx.xx.xx.xx[500] to xx.xx.xx.xx[500] (320 bytes)
06[NET] received packet: from xx.xx.xx.xx[500] to xx.xx.xx.xx[500] (80 bytes)
06[ENC] parsed CREATE_CHILD_SA response 79 [ N(INVAL_KE) ]
06[IKE] already retried with DH group ECP_521, ignore requested ECP_521
06[IKE] failed to establish CHILD_SA, keeping IKE_SA
06[IKE] CHILD_SA rekeying failed, trying again in 8 seconds

So, both sides have ECP521 (DH Group 21) configured, but somehow we get this message saying it did not accept ECP521.

Any ideas?

Thanks.

Spyro

**Comment by tobiasbrunner** [comment, marked as answer]
> So, both sides have ECP521 (DH Group 21) configured, but somehow we get this message saying it did not accept ECP521.

Check the log on the other peer. It might not actually be able to to use ECP-521 (or there is a problem when it does). If it doesn't have any other DH groups to fallback to (the initiator didn't propose any other), `INVAL_KE` with the same, original DH group could be the result.

**Reply by cpir0** [reply]
Hi Tobias and thanks for the reply! We actually changed the DH group to 24 (modp2048s256) for phase 2 and the rekey now goes through without problems. The Cisco ASA site probably had an issue with DH 21.

**Reply by cpir0** [reply]
One question, when force rekeying with "ipsec stroke rekey xx-xx-xx{xxxxx}" it goes through the same procedure that it will go through when actually rekeying when lifetime is over right?

Because that is how we tested.

**Reply by tobiasbrunner** [reply]
Yes, it's exactly the same.

**Reply by cpir0** [reply]
Thanks Tobias. Seems that the issue has been resolved! Hopefully if others stumble upon the same issue they will be able to find this discussion.
